spec/views/accounts/_work_details.html.erb_spec.rb

```diff
@@ -0,0 +1,550 @@
+          course: ehco_course,
+          raw_application_data: {
+            "npqh_status" => "completed_npqh_not_in_role",
+            "ehco_headteacher" => "no",
+          },
+        )
+        render partial: "accounts/work_details", locals: { application: application }
+      end
+
+      it "does not show first 5 years of headship" do
+        expect(subject).not_to have_css(".govuk-summary-list__key", text: "First 5 years of headship")
+      end
+    end
+  end
+
+  describe "NPQLPM course specific fields" do
+    context "when course is NPQLPM" do
+      let(:npqlpm_course) { create(:course, :leading_primary_mathematics) }
+
+      before do
+        application.update!(
+          course: npqlpm_course,
+          raw_application_data: {
+            "maths_eligibility_teaching_for_mastery" => "yes",
+          },
+        )
+        render partial: "accounts/work_details", locals: { application: application }
+      end
+
+      it "shows maths teaching for mastery question" do
+        expect(subject).to have_css(".govuk-summary-list__key", text: "Completed one year of the primary maths Teaching for Mastery programme?")
+        expect(subject).to have_css(".govuk-summary-list__value")
+      end
+    end
+
+    context "when course is NPQLPM with 'no' answer" do
+      let(:npqlpm_course) { create(:course, :leading_primary_mathematics) }
+
+      before do
+        application.update!(
+          course: npqlpm_course,
+          raw_application_data: {
+            "maths_eligibility_teaching_for_mastery" => "no",
+            "maths_understanding_of_approach" => "some_understanding",
+          },
+        )
+        render partial: "accounts/work_details", locals: { application: application }
+      end
+
+      it "shows understanding of approach instead" do
+        expect(subject).to have_css(".govuk-summary-list__key", text: "Completed one year of the primary maths Teaching for Mastery programme?")
+        expect(subject).to have_css(".govuk-summary-list__value")
+      end
+    end
+  end
+
+  describe "NPQS course specific fields" do
+    context "when course is NPQS" do
+      let(:npqs_course) { create(:course, :senco) }
+
+      before do
+        application.update!(
+          course: npqs_course,
+          raw_application_data: {
+            "senco_in_role_status" => true,
+            "senco_start_date" => "2023-01-15",
+          },
+        )
+        render partial: "accounts/work_details", locals: { application: application }
+      end
+
+      it "shows SENCO role with start date" do
+        expect(subject).to have_css(".govuk-summary-list__value", text: /Yes – since/)
+      end
+    end
+
+    context "when course is NPQS but not in role" do
+      let(:npqs_course) { create(:course, :senco) }
+
+      before do
+        application.update!(
+          course: npqs_course,
+          raw_application_data: {
+            "senco_in_role_status" => false,
+            "senco_in_role" => "no",
```

Comment:
```suggestion
            "senco_in_role" => "no_i_do_not_plan_to_be_a_SENCO",
```
this is not a valid senco_in_role option